File: kernel/web/src/types/status.ts
export interface StatusResponse {
  kernel: {
    status: string
    uptime_seconds: number
    version: string
  }
  jodo: {
    status: 'running' | 'starting' | 'unhealthy' | 'dead' | 'rebirthing' | string
    pid: number
    galla: number
    phase: 'booting' | 'thinking' | 'sleeping' | string
    uptime_seconds: number
    last_health_check: string
    health_check_ok: boolean
    restarts_today: number
    current_git_tag: string
    current_git_hash: string
  }
  database: {
    status: string
    memories_stored: number
  }
}

export interface ProviderBudget {
  monthly_budget: number
  spent_this_month: number
  remaining: number
  emergency_reserve: number
  available_for_normal_use: number
}

export interface BudgetResponse {
  providers: Record<string, ProviderBudget>
  total_spent_today: number
  budget_resets: string
}

File: kernel/internal/api/status.go
package api

import (
	"net/http"
	"time"

	"github.com/gin-gonic/gin"
)

// KernelStartTime is set at boot.
var KernelStartTime time.Time

func (s *Server) handleStatus(c *gin.Context) {
	jodoStatus := s.Process.GetStatus()

	// Get git info
	gitHash, _ := s.Git.CurrentHash()
	gitTag, _ := s.Git.CurrentTag()

	// Memory count
	memCount, _ := s.Memory.Count()

	c.JSON(http.StatusOK, gin.H{
		"kernel": gin.H{
			"status":         "running",
			"uptime_seconds": int(time.Since(KernelStartTime).Seconds()),
			"version":        "1.0.0",
		},
		"jodo": gin.H{
			"status":           jodoStatus.Status,
			"pid":              jodoStatus.PID,
			"galla":            jodoStatus.Galla,
			"phase":            jodoStatus.Phase,
			"uptime_seconds":   s.Process.UptimeSeconds(),
			"last_health_check": jodoStatus.LastHealthCheck.Format(time.RFC3339),
			"health_check_ok":  jodoStatus.HealthCheckOK,
			"restarts_today":   jodoStatus.RestartsToday,
			"current_git_tag":  gitTag,
			"current_git_hash": gitHash,
		},
		"database": gin.H{
			"status":          "connected",
			"memories_stored": memCount,
		},
	})
}
